src/nginx.py

```diff
@@ -161,15 +163,43 @@
     },
 ]
 
+LOCATIONS_BASIC: List[Dict] = [
+    {
+        "directive": "location",
+        "args": ["=", "/"],
+        "block": [
+            {"directive": "return", "args": ["200", "'OK'"]},
+            {"directive": "auth_basic", "args": ["off"]},
+        ],
+    },
+    {  # Location to be used by nginx-prometheus-exporter
+        "directive": "location",
+        "args": ["=", "/status"],
+        "block": [
+            {"directive": "stub_status", "args": []},
+        ],
+    },
+]
+
 
 class Nginx:
     """Helper class to manage the nginx workload."""
 
     config_path = NGINX_CONFIG
 
-    def __init__(self, cluster_provider: MimirClusterProvider, server_name: str):
+    def __init__(self, charm: CharmBase, cluster_provider: MimirClusterProvider, server_name: str):
+        self._charm = charm
         self.cluster_provider = cluster_provider
         self.server_name = server_name
+        self._container = self._charm.unit.get_container("nginx")
+
+    def configure_pebble_layer(self) -> None:
+        """Configure pebble layer."""
+        self._container.push(
+            self.config_path, self.config(tls=self._charm._is_cert_available), make_dirs=True  # type: ignore
```

Comment:
Would it make sense to pass `_is_cert_available` as an arg to `configure_pebble_layer`?

Reply:
Yes, it is better.
I will add a `_is_tls_ready` property... the `_is_cert_available` does not check the certs are in the container...